src/daft-core/src/array/ops/cast.rs

```diff
@@ -430,6 +430,32 @@ impl DurationArray {
             _ => arrow_cast(&self.physical, dtype),
         }
     }
+
+    pub fn cast_to_days(&self) -> DaftResult<Series> {
```

Comment:
Why not return an Int32Array here so it's explicit? No need for the loss of information if we know exactly what type it is.